actors/builtin/verifreg/verified_registry_state.go

```diff
@@ -37,8 +52,79 @@ func ConstructState(store adt.Store, rootKeyAddress addr.Address) (*State, error
 	}
 
 	return &State{
-		RootKey:         rootKeyAddress,
-		Verifiers:       emptyMapCid,
-		VerifiedClients: emptyMapCid,
+		RootKey:                  rootKeyAddress,
+		Verifiers:                emptyMapCid,
+		VerifiedClients:          emptyMapCid,
+		RemoveDataCapProposalIDs: emptyMapCid,
 	}, nil
 }
+
+// A verifier who wants to send/agree to a RemoveDataCapRequest should sign a RemoveDataCapProposal and send the signed proposal to the root key holder.
+type RemoveDataCapProposal struct {
+	// VerifiedClient is the client address to remove the DataCap from
+	// The address must be an ID address
+	VerifiedClient addr.Address
+	// DataCapAmount is the amount of DataCap to be removed from the VerifiedClient address
+	DataCapAmount DataCap
+	// RemovalProposalID is the counter of the proposal sent by the Verifier for the VerifiedClient
+	RemovalProposalID RmDcProposalID
+}
+
+// A verifier who wants to submit a request should send their RemoveDataCapRequest to the RKH.
+type RemoveDataCapRequest struct {
+	// Verifier is the verifier address used for VerifierSignature.
+	// The address can be address.SECP256K1 or address.BLS
+	Verifier addr.Address
+	// VerifierSignature is the Verifier's signature over a RemoveDataCapProposal
+	VerifierSignature crypto.Signature
+}
+
+func isVerifier(rt runtime.Runtime, st State, address addr.Address) (bool, exitcode.ExitCode, error) {
+
+	verifiers, err := adt.AsMap(adt.AsStore(rt), st.Verifiers, builtin.DefaultHamtBitwidth);
+	if err != nil {
+		return false, exitcode.ErrIllegalState, xerrors.Errorf("failed to load verifiers.")
+	}
+
+	nina, err := verifiers.Get(abi.AddrKey(address), nil)
+	if err != nil {
+		return false, exitcode.ErrIllegalState, xerrors.Errorf("failed to load verifier %v.", address)
+	}
+	if !nina {
+		rt.Abortf(exitcode.ErrNotFound, "%v is not a verifier", address)
+	}
+
+	return true, 0, nil
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// State utility functions
+////////////////////////////////////////////////////////////////////////////////
+func removeDataCapRequestIsValidOrAbort(rt runtime.Runtime, request RemoveDataCapRequest, id RmDcProposalID, toRemove DataCap, clientRemoved address.Address) {
+	proposal := RemoveDataCapProposal{
+		RemovalProposalID: id,
+		DataCapAmount:     toRemove,
+		VerifiedClient:    clientRemoved,
+	}
+	buf := bytes.Buffer{}
+	if err := proposal.MarshalCBOR(&buf); err != nil {
+		rt.Abortf(exitcode.ErrSerialization, "remove datacap request signature validation failed to marshal request: %s", err)
+	}
+
+	if err := rt.VerifySignature(request.VerifierSignature, request.Verifier, buf.Bytes()); err != nil {
+		rt.Abortf(exitcode.SysErrorIllegalArgument, "remove datacap request signature is invalid: %s", err)
```

Comment:
```suggestion
		rt.Abortf(exitcode.ErrIllegalArgument, "remove datacap request signature is invalid: %s", err)
```

This should be an actor error, not a system error